### PR TITLE
This adds a search sync feature to reformat data to elasticsearch

### PR DESCRIPTION
**Pull request by dmlb2000**
### Description

This adds a search sync feature to create an index in elasticsearch with formatted data that is more searchable for elasticsearch.

### Issues Resolved

N/A
### Check List

- [ ] All tests pass. See <https://github.com/pacifica/pacifica-docs/blob/master/development/TESTING.md>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

File: policy/search_render.py [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+        )
+    ]
+
+
+class SearchRender(object):
+    """Search render class to contain methods."""
+
+    obj_cache = LimitedSizeDict(size_limit=CACHE_SIZE)
+    render_data = {
+        'instruments': {
```
Comment (makopov) [inline policy/search_render.py]: Does it make sense to put the crosstables in here as well?

File: policy/search_sync.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+    return esclient
+
+
+def try_es_connect(attempts=0):
+    """Recursively try to connect to elasticsearch."""
+    try:
+        cli = es_client()
+        cli.info()
+    except ElasticsearchException as ex:  # pragma: no cover pulled from metadata
+        if attempts < ELASTIC_CONNECT_ATTEMPTS:
```
Comment (makopov) [inline policy/search_sync.py]: This is confusing. `attempts` is an argument which means the class interacting with this interface can specify the number of times to attempt a reconnect (presumably). However, you're treating it as an incremental variable and not a constant

File: policy/test/test_connect.py [hunk 3]
```diff
@@ -17,9 +17,7 @@ def test_meta_connect(self):
         httpretty.register_uri(httpretty.GET, METADATA_STATUS_URL,
                                body=dumps([]),
                                content_type='application/json')
-        # pylint: disable=protected-access
-        Root._try_meta_connect(0)
-        # pylint: enable=protected-access
+        Root.try_meta_connect(0)
```
Comment (makopov) [inline policy/test/test_connect.py]: Just a nitpick here but the default argument is already 0

**Review by makopov — CHANGES_REQUESTED** [review]

File: policy/search_render.py (hunk 1 above)
Comment (dmlb2000) [reply policy/search_render.py]: This is a function of how elasticsearch works, the cross tables are embedded as a list in an attribute of each object.

File: policy/search_sync.py (hunk 2 above)
Comment (dmlb2000) [reply policy/search_sync.py]: Yes, the recursion is what you are missing. Later in the block I call the method again with the attempts increased by 1.

File: policy/test/test_connect.py (hunk 3 above)
Comment (dmlb2000) [reply policy/test/test_connect.py]: Yes, this is a test and just to make sure we are covering a specific part of the metadata connect logic. So I made sure the attempts was set to zero so we could cover that.

**Review by makopov — APPROVED** [review]

**Pull request merged**